app/src/lib/stores/app-store.ts

```diff
@@ -4604,24 +4624,39 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
-  public _setDivergingBranchBannerVisibility(
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public _updateDivergingBranchBannerState(
     repository: Repository,
-    visible: boolean
+    divergingBranchBannerState: Partial<IDivergingBranchBannerState>
   ) {
-    const state = this.repositoryStateCache.get(repository)
-    const { compareState } = state
-
-    if (compareState.isDivergingBranchBannerVisible !== visible) {
-      this.repositoryStateCache.updateCompareState(repository, () => ({
-        isDivergingBranchBannerVisible: visible,
-      }))
+    const currentBannerState = this.repositoryStateCache.get(repository)
+      .compareState.divergingBranchBannerState
+
+    const newBannerState = {
+      isPromptVisible:
+        divergingBranchBannerState.isPromptVisible !== undefined
+          ? divergingBranchBannerState.isPromptVisible
+          : currentBannerState.isPromptVisible,
+      isNudgeVisible:
+        divergingBranchBannerState.isNudgeVisible !== undefined
+          ? divergingBranchBannerState.isNudgeVisible
+          : currentBannerState.isNudgeVisible,
+      isPromptDismissed:
+        divergingBranchBannerState.isPromptDismissed !== undefined
+          ? divergingBranchBannerState.isPromptDismissed
+          : currentBannerState.isPromptDismissed,
+    }
```

Comment:
@rafeca we talked about this already, just recording this here.

I think we can replace this block with a couple object spreads— 
```suggestion
    const newBannerState = { ...currentBannerState, ...divergingBranchBannerState }
```